runtimes/eden/src/lib.rs

```diff
@@ -0,0 +1,371 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#![cfg_attr(not(feature = "std"), no_std)]
+// `construct_runtime!` does a lot of recursion and requires us to increase the limit to 256.
+#![recursion_limit = "256"]
+
+// Make the WASM binary available.
+#[cfg(feature = "std")]
+include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+/// Wasm binary unwrapped. If built with `SKIP_WASM_BUILD`, the function panics.
+#[cfg(feature = "std")]
+pub fn wasm_binary_unwrap() -> &'static [u8] {
+    WASM_BINARY.expect(
+        "Development wasm binary is not available. This means the client is \
+        built with `SKIP_WASM_BUILD` flag and it is only usable for \
+        production chains. Please rebuild with the flag disabled.",
+    )
+}
+
+use frame_support::construct_runtime;
+use pallet_transaction_payment::{FeeDetails, RuntimeDispatchInfo};
+use primitives::{AccountId, Balance, BlockNumber, Index, Signature};
+pub use primitives::{AuraId, ParaId};
+use sp_core::OpaqueMetadata;
+#[cfg(any(feature = "std", test))]
+pub use sp_runtime::BuildStorage;
+use sp_runtime::{
+    generic,
+    traits::{BlakeTwo256, Block as BlockT},
+    transaction_validity::{TransactionSource, TransactionValidity},
+    ApplyExtrinsicResult,
+};
+use sp_std::prelude::*;
+use sp_version::RuntimeVersion;
+
+pub mod constants;
+mod implementations;
+mod pallets_consensus;
+mod pallets_governance;
+mod pallets_nodle;
+mod pallets_parachain;
+mod pallets_system;
+mod pallets_util;
+mod version;
+
+pub use pallets_consensus::SessionKeys;
+#[cfg(feature = "std")]
+pub use version::native_version;
+pub use version::VERSION;
+
+construct_runtime! {
+    pub enum Runtime where
+        Block = Block,
+        NodeBlock = primitives::Block,
+        UncheckedExtrinsic = UncheckedExtrinsic
+    {
+        // System
+        System: frame_system::{Pallet, Call, Storage, Config, Event<T>} = 0,
+        Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent} = 1,
+        Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>} = 2,
+        TransactionPayment: pallet_transaction_payment::{Pallet, Storage} = 3,
+        RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 4,
+
+        // Governance
+        TechnicalCommittee: pallet_collective::<Instance2>::{Pallet, Call, Storage, Origin<T>, Event<T>, Config<T>} = 5,
+        TechnicalMembership: pallet_membership::<Instance1>::{Pallet, Call, Storage, Event<T>, Config<T>} = 6,
```

Comment:
we can remove these two

Reply:
@ETeissonniere , for me it occurs like, we lose control over addition & removal of allocations oracle members, is it OK ?

```
    #[pallet::storage]
    #[pallet::getter(fn oracles)]
    pub type Oracles<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;

```

Reply:
We can use `EnsureRoot` for controlling who can add oracles or remove them

Reply:
Make sense, looks good.

Reply:
Done @Eliott, added AllocationsOracles back, and its working for sudo origin